pkg/ebpf/policy_manager.go

```diff
@@ -10,47 +10,129 @@ import (
 	return &policyManager{
 		mutex: sync.Mutex{},
-		rules: make(map[events.ID]uint64),
+		rules: make(map[events.ID]*eventState),
+	}
+}
+
+// IsEnabled tests if a event, or a policy per event is enabled (in the future it will also check if a policy is enabled)
+// TODO: add metrics about an event being enabled/disabled, or a policy being enabled/disabled?
+func (pm *policyManager) IsEnabled(matchedPolicies uint64, ruleId events.ID) bool {
+	pm.mutex.Lock()
+	defer pm.mutex.Unlock()
+
+	if !pm.isEventEnabled(ruleId) {
+		return false
 	}
+
+	return pm.isRuleEnabled(matchedPolicies, ruleId)
 }
 
 // IsRuleEnabled returns true if a given event policy is enabled for a given rule
 func (pm *policyManager) IsRuleEnabled(matchedPolicies uint64, ruleId events.ID) bool {
 	pm.mutex.Lock()
 	defer pm.mutex.Unlock()
 
-	policyMask, ok := pm.rules[ruleId]
+	return pm.isRuleEnabled(matchedPolicies, ruleId)
+}
+
+// not synchronized, use IsRuleEnabled instead
+func (pm *policyManager) isRuleEnabled(matchedPolicies uint64, ruleId events.ID) bool {
+	state, ok := pm.rules[ruleId]
+	if !ok {
+		return false
+	}
+
+	return state.policyMask&matchedPolicies != 0
+}
+
+// IsEventEnabled returns true if a given event policy is enabled for a given rule
+func (pm *policyManager) IsEventEnabled(evenId events.ID) bool {
+	pm.mutex.Lock()
+	defer pm.mutex.Unlock()
+
+	return pm.isEventEnabled(evenId)
+}
+
+// not synchronized, use IsEventEnabled instead
+func (pm *policyManager) isEventEnabled(evenId events.ID) bool {
+	state, ok := pm.rules[evenId]
 	if !ok {
 		return false
 	}
 
-	return policyMask&matchedPolicies != 0
+	return state.enabled
 }
 
 // EnableRule enables a rule for a given event policy
 func (pm *policyManager) EnableRule(policyId int, ruleId events.ID) {
 	pm.mutex.Lock()
 	defer pm.mutex.Unlock()
 
-	policyMask := pm.rules[ruleId]
-	utils.SetBit(&policyMask, uint(policyId))
+	state, ok := pm.rules[ruleId]
+	if !ok {
+		// if you enabling/disabling a rule for an event that
+		// was not enabled/disabled yet, we assume the event should be enabled
+		state = &eventState{enabled: true}
+	}
+
+	utils.SetBit(&state.policyMask, uint(policyId))
 
-	pm.rules[ruleId] = policyMask
+	pm.rules[ruleId] = state
 }
 
 // DisableRule disables a rule for a given event policy
 func (pm *policyManager) DisableRule(policyId int, ruleId events.ID) {
 	pm.mutex.Lock()
 	defer pm.mutex.Unlock()
 
-	policyMask := pm.rules[ruleId]
-	utils.ClearBit(&policyMask, uint(policyId))
+	state, ok := pm.rules[ruleId]
+	if !ok {
+		// if you enabling/disabling a rule for an event that
+		// was not enabled/disabled yet, we assume the event should be enabled
+		state = &eventState{enabled: true}
+	}
+
+	utils.ClearBit(&state.policyMask, uint(policyId))
+
+	pm.rules[ruleId] = state
+}
+
+// EnableEvent enables a given event
+func (pm *policyManager) EnableEvent(eventId events.ID) {
+	pm.mutex.Lock()
+	defer pm.mutex.Unlock()
+
+	state, ok := pm.rules[eventId]
```

Comment:
So how will this code look like in the future when we will have rule ids different than event id?
We will not be able to use pm.rules map anymore, right?

Reply:
Probably, but it is an internal structure we should be able to refactor it without affecting anything that is consuming it. Right? It is encapsulated.